src/unity.c

```diff
@@ -203,19 +188,21 @@ void UnityPrintNumberUnsigned(const UNITY_UINT number)
 /*-----------------------------------------------*/
 void UnityPrintNumberHex(const UNITY_UINT number, const char nibbles_to_print)
 {
-    UNITY_UINT nibble;
+    int nibble;
     char nibbles = nibbles_to_print;
+    if ((unsigned)nibbles > 2 * sizeof number) nibbles = 2 * sizeof number;
 
     while (nibbles > 0)
     {
-        nibble = (number >> (--nibbles << 2)) & 0x0000000F;
+        nibbles--;
+        nibble = (number >> (nibbles * 4)) & 0x0F;
         if (nibble <= 9)
         {
-            UNITY_OUTPUT_CHAR((char)('0' + nibble));
+            UNITY_OUTPUT_CHAR('0' + nibble);
         }
         else
         {
-            UNITY_OUTPUT_CHAR((char)('A' - 10 + nibble));
+            UNITY_OUTPUT_CHAR('A' - 10 + nibble);
```

Comment:
...changing it back.

What I was thinking:
Since the prototype is:
```C
void UNITY_OUTPUT_CHAR(int);
```
I guessed that the narrowing cast wasn't required. I do test these things with several compilers, but it's easy to miss a warning. Thanks for catching it.